packages/web-config-server/src/apiV1/organisationUnit.js

```diff
@@ -12,12 +12,46 @@ const translateDescendantForFrontEnd = (descendant, childIdToParentId, entityIdT
   parent: entityIdToCode[childIdToParentId[descendant.id]],
 });
 
+async function fetchEntitiesWithProjectAccess(req, entities, userGroups) {
+  return Promise.all(
+    entities.map(async ({ id, name, code, bounds }) => ({
+      id,
+      name,
+      code,
+      bounds,
+      hasAccess: await Promise.all(userGroups.map(u => req.userHasAccess(code, u))),
+    })),
+  );
+}
+
 export default class extends RouteHandler {
   static PermissionsChecker = PermissionsChecker; // checks the user has access to requested entity
 
   async buildResponse() {
     const { includeCountryData } = this.query;
     const project = await this.fetchProject();
+
+    // If this is a project entity, get entitiesWithAccess and set them on the entity
+    if (this.entity.isProject()) {
+      const orgUnits = await this.entity.getNearestOrgUnitDescendants();
+
+      const accessByEntity = await Promise.all(
```

Comment:
The `xByY` naming pattern is usually used for objects where `y` is the key and `x` is the value. This would be better named something like `entitiesWithAccessInfo` as it's an array.

This probably comes from the `project` endpoint, so I realise you aren't introducing the specific name here!